Realm.Shared/linq/RealmResultsVisitor.cs

```diff
@@ -179,47 +203,73 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
                             firstRowPtr = rh.GetRow(0);
                         }
                     }
-                    if (firstRowPtr == IntPtr.Zero)
+                    if (firstRowPtr != IntPtr.Zero)
+                        return Expression.Constant(_realm.MakeObjectForRow(_metadata, firstRowPtr));
+                    if (m.Method.Name ==nameof(Queryable.First))
                         throw new InvalidOperationException("Sequence contains no matching element");
-                    return Expression.Constant(_realm.MakeObjectForRow(_metadata, firstRowPtr));
+                    Debug.Assert (m.Method.Name == nameof(Queryable.FirstOrDefault));
+                    return Expression.Constant(null);
+                }
+                if (m.Method.Name == nameof(Queryable.DefaultIfEmpty))
```

Comment:
Oh, makes sense. But maybe just comment out the whole block so that it's still there in source form, but if anybody calls `DefaultIfEmpty` they would get a not supported exception, instead of undefined behavior.